torch_xla/csrc/tensor_methods.cpp

```diff
@@ -1678,26 +1678,6 @@ XLATensor XLATensor::linspace(const at::Scalar& start, const at::Scalar& end,
       element_type);
 }
 
-XLATensor XLATensor::log(const XLATensor& input) {
-  // Here we explictly pass c10::nullopt as logical_element_type because
-  // otherwise result will inherit the input's logical_element_type. In the
-  // case of log(int) -> float, we want to derive the dtype from IR value
-  // instead of input's logical_element_type.
```

Comment:
hmm, I think this should be fine. Now with the codegen we should compute the `logical_element_type` from computed `torch::lazy::shape`

Reply:
I added type casting to the lowering stage. Letting CI run the test as local testing is broken at the moment.